systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -1671,6 +1675,145 @@ void testRegenerateCertExternalAddressChange() throws InterruptedException {
+
+        LOGGER.info("Getting labels from stateful set resource");
+        StatefulSet statefulSet = kubeClient().getStatefulSet(kafkaClusterName(CLUSTER_NAME));
+        labels = statefulSet.getSpec().getTemplate().getMetadata().getLabels();
+        LOGGER.info("Verifying default labels in the KafkaCR");
+
+        assertEquals(labelValues[0], labels.get(labelKeys[0]));
+        assertEquals(labelValues[1], labels.get(labelKeys[1]));
+
+        String newLabelKey = "label-name-3";
+        String newLabelValue = "name-of-the-label-3";
+
+        LOGGER.info("Edit kafka labels in Kafka CR");
+        replaceKafkaResource(CLUSTER_NAME, resource -> {
+            resource.getMetadata().getLabels().put(labelKeys[0], labelNewValues[0]);
+            resource.getMetadata().getLabels().put(labelKeys[1], labelNewValues[1]);
+            resource.getMetadata().getLabels().put(newLabelKey, newLabelValue);
+        });
+
+        labels.put(labelKeys[0], labelNewValues[0]);
+        labels.put(labelKeys[1], labelNewValues[1]);
+        labels.put(newLabelKey, newLabelValue);
+
+        LOGGER.info("Waiting for kafka service labels changed {}", labels);
+        StUtils.waitForKafkaServiceLabelsChange(brokerServiceName, labels);
+
+        LOGGER.info("Verifying kafka labels via services");
+        Service service = kubeClient().getService(brokerServiceName);
+
+        assertEquals(labelNewValues[0], service.getMetadata().getLabels().get(labelKeys[0]));
+        assertEquals(labelNewValues[1], service.getMetadata().getLabels().get(labelKeys[1]));
+        assertEquals(newLabelValue, service.getMetadata().getLabels().get(newLabelKey));
+
+        LOGGER.info("Waiting for kafka config map labels changed {}", labels);
+        StUtils.waitForKafkaConfigMapLabelsChange(configMapName, labels);
+
+        LOGGER.info("Verifying kafka labels via config maps");
+        ConfigMap configMap = kubeClient().getConfigMap(configMapName);
+
+        assertEquals(labelNewValues[0], configMap.getMetadata().getLabels().get(labelKeys[0]));
+        assertEquals(labelNewValues[1], configMap.getMetadata().getLabels().get(labelKeys[1]));
+        assertEquals(newLabelValue, configMap.getMetadata().getLabels().get(newLabelKey));
+
+        LOGGER.info("Waiting for kafka stateful set labels changed {}", labels);
+        StUtils.waitForKafkaStatefulSetLabelsChange(kafkaClusterName(CLUSTER_NAME), labels);
+
+        LOGGER.info("Verifying kafka labels via stateful set");
+        statefulSet = kubeClient().getStatefulSet(kafkaClusterName(CLUSTER_NAME));
+        labels = statefulSet.getSpec().getTemplate().getMetadata().getLabels();
+
+        assertEquals(labelNewValues[0], labels.get(labelKeys[0]));
+        assertEquals(labelNewValues[1], labels.get(labelKeys[1]));
+        assertEquals(newLabelValue, labels.get(newLabelKey));
+
+        StUtils.waitForReconciliation(testClass, testName, NAMESPACE);
```

Comment:
This seems to wait for the periodical reconciliation. I think that is hiding two problems:
* It seems to check the reconciliation start which doesn't guarantee the pods were rolled already
* Updating the resource metadata should trigger reconciliation on its own. No need to wait for periodic reconciliation.

Reply:
I added waitTillSsHasRolled method, which should solve these problems.